Microsoft.Research/ManagedContract.Setup/buildMSI10.xml

```diff
@@ -370,8 +371,33 @@
 
     ==========================================================-->
   <Target Name="MergeExes"
-    DependsOnTargets="ILMergeFoxtrot;ILMergeAsmMeta;ILMergeClousot;ILMergeCCDoc"/>
+    DependsOnTargets="PrepareMerges;ExecuteMerges"/>
 
+  <Target Name="PrepareMerges" BeforeTargets="ExecuteMerges"
+    DependsOnTargets="RestoreToolPackages;PrepareILMerge"/>
+	
+  <Target Name="RestoreToolPackages">
+    <Exec Command="&quot;$(SolutionDir).NuGet\NuGet.exe&quot; restore &quot;$(SolutionDir)packages.config&quot; -PackagesDirectory &quot;$(SolutionDir)packages&quot;" />

```

Comment:
@hubuk I want to see this pull request merged. To get us back on track to do so, can you reiterate the reasons why you don't want to move the call to `nuget restore` outside of this file? My reasons in favor of moving it are the following:

1. Command line restore matches the behavior of automatic package restore in Visual Studio. The restore operation is completed before entering the MSBuild process.
2. Command line restore works in all scenarios, and is an easily recognized pattern. IMO, it leaves no confusion regarding the manner and timing in which packages are restored during a command line build. Restoring packages in an MSBuild target requires separate consideration of multiple entry points (the selected build targets) and declared dependencies, and runs a risk that a future configuration might force us back to command line restoration anyway.

Summary: I believe command line restore prior to MSBuild is simpler and more robust in the face of unknown project developments in the future. I have higher confidence that we will be happy with it as a long-term solution for integrating NuGet packages into the build.